staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -434,14 +434,18 @@ func (r *crdHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	var handlerFunc http.HandlerFunc
-	subresources, err := apiextensionshelpers.GetSubresourcesForVersion(crd, requestInfo.APIVersion)
-	if err != nil {
-		utilruntime.HandleError(err)
-		responsewriters.ErrorNegotiated(
-			apierrors.NewInternalError(fmt.Errorf("could not properly serve the subresource")),
-			Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req,
-		)
-		return
+	var subresources *apiextensionsv1.CustomResourceSubresources
+	// Subresources (scale, status) are not applicable for wildcard partial metadata requests
+	if !wildcardPartialMetadata {
+		subresources, err = apiextensionshelpers.GetSubresourcesForVersion(crd, requestInfo.APIVersion)
+		if err != nil {
+			utilruntime.HandleError(err)
+			responsewriters.ErrorNegotiated(
+				apierrors.NewInternalError(fmt.Errorf("could not properly serve the subresource")),
+				Codecs, schema.GroupVersion{Group: requestInfo.APIGroup, Version: requestInfo.APIVersion}, w, req,
+			)
+			return
+		}
 	}
 	switch {
 	case subresource == "status" && subresources != nil && subresources.Status != nil:
```

Comment:
won't this panic if `subresources` is `nil` - for `wildcardPartialMetadata = true` ?

Reply:
No, it has a `subresources != nil` guard. Also, it's expected that a CRD may have 0 subresources. We are just forcing wildcard partial metadata requests to always have no subresources.